packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -136,14 +135,17 @@ export class ProfilesCache {
      *
      * @returns {void}
      */
-    public registerCustomProfilesType(profileTypeName: string): void {
-        const exists = fs.existsSync(path.posix.join(`${os.homedir()}/.zowe/profiles/${profileTypeName}`));
-        if (!exists) {
-            throw new Error(
-                `Zowe Explorer Profiles Cache error: Tried to register a custom profile type named: ${profileTypeName} that does not yet exist. Extenders must call "zoweExplorerApi.getExplorerExtenderApi().initForZowe()" first.`
-            );
-        }
-        this.allExternalTypes.add(profileTypeName);
+    public registerCustomProfilesType(profileTypeName: string) {
+        const existsV1Profile = fs.existsSync(path.posix.join(`${os.homedir()}/.zowe/profiles/${profileTypeName}`));
+        this.getProfileInfo().then((res) => {
```

Comment:
This call requires a full `then..catch` block according to the `eslint` rules again. So I would suggest to add simple:
```
.then(...)
.catch((error) => { throw error; });
```
to fix the build.

Reply:
An alternative solution might be to convert `registerCustomProfilesType` function into `async` function to fully use `await` operator. But since this is a public API, you may want to introduce a new `asyncRegisterCustomProfilesType` call and deprecate the existing one.

This approach would be preferable, because otherwise we are loosing error throwing from `registerCustomProfilesType` function itself (they will stuck in callbacks).

Reply:
Thanks Roman your inputs much appreciated :)